### PR TITLE
feat!: `x/did` and `x/resource`  module changes according to sdk v0.47

### PR DESCRIPTION
**Pull request by deepan95dev**
Closes: #687

### CODE REVIEW

File: scripts/protocgen.sh [hunk 1]
```diff
@@ -16,10 +15,13 @@ for proto_dir in $proto_dirs; do
   done
 done
 
+ls -l
 cd ..
+ls -l
```
Comment (akhilkumarpilli) [inline scripts/protocgen.sh:20]:
```suggestion
cd ..
```

File: x/did/keeper/querier.go
Comment (akhilkumarpilli) [inline x/did/keeper/querier.go:1]: Remove this file if not needed

File: x/did/keeper/querier_all_diddoc_versions_metadata.go
Comment (akhilkumarpilli) [inline x/did/keeper/querier_all_diddoc_versions_metadata.go:1]: Please make sure that these queries are implemented in GRPC QueryServer.

File: x/did/module.go
Comment (akhilkumarpilli) [inline x/did/module.go:1]: Please add AppModule missing methods

File: x/did/keeper/querier_all_diddoc_versions_metadata.go
Comment (atheeshp) [reply x/did/keeper/querier_all_diddoc_versions_metadata.go:1]: yes they are available, rechecked!

**Review by akhilkumarpilli — APPROVED** [review]
lgtm

**Pull request merged**